### PR TITLE
Implement topologySpreadConstraints

### PR DESCRIPTION
**Pull request by runningman84**
this is useful for ha purpose

### CODE REVIEW

File: helm-chart/kubemod/templates/deployment.yaml [hunk 1]
```diff
@@ -83,3 +83,16 @@ spec:
...
+            {{- include "kubemod.selectorLabels" . | nindent 12 }}
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: ScheduleAnyway
+      - labelSelector:
+          matchLabels:
+            {{- include "kubemod.selectorLabels" . | nindent 12 }}
+        maxSkew: 1
+        topologyKey: kubernetes.io/hostname
+        whenUnsatisfiable: DoNotSchedule
```
Comment (vassilvk) [inline helm-chart/kubemod/templates/deployment.yaml]: Not sure I agree with `DoNotSchedule` for this topology key.
We should be able to deploy multiple instances of KubeMod to the same node in order to provide high availability.

**Comment by runningman84** [comment]
@vassilvk  would you mind merging this?

**Comment by vassilvk** [comment]
Hi @runningman84, did you see my comment regarding `DoNotSchedule` for topology key `kubernetes.io/hostname`?

**Comment by runningman84** [comment]
No I have not seen any comment. The reason for do not schedule on the same node is because it creates a false sense of security and increases outages . I would rather have a second pod in pending state which would result in karpenter or cluster autoscaler creating another node.

**Comment by vassilvk** [comment]
Hi @runningman84 - yes, this makes sense.
Please fix the failing lint test by bumping the chart version and we can go ahead with the merge.

File: helm-chart/kubemod/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: kubemod
 description: KubeMod is a universal Kubernetes mutating operator.
-version: 0.4.0
+version: 0.4.1
```
Comment (vassilvk) [inline helm-chart/kubemod/Chart.yaml]: Hi @runningman84, please set the version to 0.5.1

**Review by vassilvk — CHANGES_REQUESTED** [review]

File: helm-chart/kubemod/templates/deployment.yaml (hunk 1 above)
Comment (vassilvk) [reply helm-chart/kubemod/templates/deployment.yaml]: Resolving this as I agree with @runningman84‘s notion regarding HA and forcing provisioner node spin up through unschedulable pod.

**Pull request merged**